### PR TITLE
fix(utils): only set retryWrites to true for valid operations

### PR DESCRIPTION
**Pull request by kvwalker**
Fixes NODE-1641

### CODE REVIEW

**Comment by mbroadst** [comment]
@daprahamian agreed that we have finally realized the extent of my initial comment about this being the wrong location for this. 

@kvwalker we are presently checking for whether a command supports retryability [here](https://github.com/mongodb-js/mongodb-core/blob/master/lib/topologies/replset.js#L1280-L1284). Was this just not caught there?

File: lib/utils.js [hunk 1]
```diff
@@ -441,25 +441,35 @@ const executeOperation = (topology, operation, args, options) => {
   });
 };
 
+/**
+ * Applies retryWrites: true to a command if retryWrites is set on the command's database.
+ *
+ * @param {object} target The target command to which we will apply retryWrites.
+ * @param {object} db The database from which we can inherit a retryWrites value.
+ */
+function applyRetryableWrites(target, db) {
```
Comment (daprahamian) [inline lib/utils.js]: This preserves our current behavior, so we don't need to decide this now. However, I wonder if we are expected to support `retryWrites` at the collection/operation level.

File: lib/utils.js [hunk 2]
```diff
@@ -441,25 +441,35 @@ const executeOperation = (topology, operation, args, options) => {
...
+}
+
 /**
  * Applies a write concern to a command based on well defined inheritance rules, optionally
  * detecting support for the write concern in the first place.
  *
  * @param {Object} target the target command we will be applying the write concern to
  * @param {Object} sources sources where we can inherit default write concerns from
  * @param {Object} [options] optional settings passed into a command for write concern overrides
+ * @param {boolean} isRetryable=false indicates if the operation is supported for retryable writes
```
Comment (daprahamian) [inline lib/utils.js]: We can get rid of this.

File: lib/utils.js (hunk 1 above)
Comment (kvwalker) [reply lib/utils.js]: According to the [spec](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst), "This option MUST NOT be configurable at the level of a database object, collection object, or at the level of an individual write operation."

Comment (daprahamian) [reply lib/utils.js]: ok, awesome!

**Comment by mathieug** [comment]
When do you plan to release 3.1.5? Will this fix be included? Thanks.

**Comment by mbroadst** [comment]
@mathieug 3.1.5 will be released this week

**Pull request merged**